File: 프로그래머스/lv3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    var answer = 0;

    const chkb = Array(n).fill(false);
    const graph = Array.from(Array(n), () => new Array());

    for (let i = 0; i < n; i += 1) {
        for (let j = 0; j < n; j += 1) {
            if (computers[i][j] == 1) {
                graph[i].push(j)
            }
        }
    }

    function bfs(x) {
        const q = []
        q.push(x)
        chkb[x] = true;
        while (q.length) {
            let nx = q.shift();
            for (const v of graph[nx]) {
                if (chkb[v] == false) {
                    q.push(v)
                    chkb[v] = true;
                }
            }
        }
    }

    for (let i = 0; i < n; i += 1) {
        if (chkb[i] == false) {
            bfs(i);
            answer += 1;
        }
    }

    return answer;
}

File: 프로그래머스/lv3/43162. 네트워크/네트워크.py
from collections import deque
def solution(n, computers):

    answer = 0
    network = [[] for _ in range(n)]
    chkb = [False] * n

    def bfs(network, start, chkb):
        q = deque()
        q.append(start)
        chkb[start] = True
        while q:
            x = q.popleft()
            for k in network[x]:
                if not chkb[k]:
                    q.append(k)
                    chkb[k] = True

    for i in range(len(computers)):
        for j in range(len(computers[0])):
            if computers[i][j] == 1:
                network[i].append(j)

    for i in range(len(computers)):
        if not chkb[i]:
            bfs(network, i, chkb)
            answer += 1

    return answer